### PR TITLE
Use CryptographicBuffer to create encryption strings instead of GUIDS.

### PR DESCRIPTION
**Pull request by jbernoudy**
Please review this change.  @smcnulty-sfdc

### CODE REVIEW

**Comment by wpeter-sfdc** [comment]
:+1:

File: SalesforceSDK/Salesforce.SDK.Core/Source/Security/HmacSHA256KeyGenerator.cs [hunk 1]
```diff
@@ -154,8 +154,21 @@ private static IBuffer GetNonce(string nonce)
...
-            salt = Guid.NewGuid().ToString();
+            password = GenerateRandomEncryptionString();
+            salt = GenerateRandomEncryptionString();
+        }
+
+        // Ensure a truly random string that is good for use in encrpytion.
+        private string GenerateRandomEncryptionString()
+        {
+            // Define the length, in bytes, of the buffer.
+            UInt32 length = 16;
```
Comment (maliroteh) [inline SalesforceSDK/Salesforce.SDK.Core/Source/Security/HmacSHA256KeyGenerator.cs]: nit: this could probably be a const, but not a big deal.

**Comment by maliroteh** [comment]
Looks good to me, with a minor nit !

**Comment by kchitalia** [comment]
Thinking like a security review person now, would it be possible to randomly generate the length?.
Looks good other wise :)

**Comment by jbernoudy** [comment]
@kchitalia Do we want to randomize the string length?  That would not be hard, but it might be unnecessary to add more complexity to the generation of a random string.

**Comment by kchitalia** [comment]
Lets check with Martin.

**Comment by kchitalia** [comment]
confirmed with Martin, this looks good. :+1:

**Comment by kchitalia** [comment]
@mvigo

**Comment by kchitalia** [comment]
@martinvigo

**Comment by martinvigo** [comment]
Looks good now!

**Comment by jbernoudy** [comment]
@smcnulty-sfdc This has been signed off on.  Can you merge it when you have time.  Thanks.

**Pull request merged**